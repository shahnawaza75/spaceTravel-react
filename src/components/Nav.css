.nav-items {
  display: flex;
  margin-right: 15px;
}

.nav-items li {
  list-style: none;
  margin-right: 20px;
}

.nav-items li:last-child {
  border-left: 1px solid #000;
  margin-right: 40px;
  padding-left: 12px;
}
