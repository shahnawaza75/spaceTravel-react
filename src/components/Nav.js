import { NavLink } from 'react-router-dom';
import './Nav.css';

const Nav = () => {
  const style = () => (
    {
      textDecoration: 'none',
      color: 'blue',
    }
  );
  return (
    <ul className="nav-items">
      <li><NavLink style={style} activeStyle={{ textDecoration: 'underline', color: 'rgb(124 166 215' }} to="/rockets">Rockets</NavLink></li>
      <li><NavLink style={style} activeStyle={{ textDecoration: 'underline', color: 'rgb(124 166 215' }} to="/missions">Missions</NavLink></li>
      <li><NavLink style={style} activeStyle={{ textDecoration: 'underline', color: 'rgb(124 166 215' }} to="/profile">My Profile</NavLink></li>
    </ul>
  );
};
export default Nav;
