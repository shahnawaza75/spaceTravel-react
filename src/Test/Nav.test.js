import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Nav from '../components/Nav';

describe('Test the navigator', () => {
  it('Navigation is rendered', () => {
    const nav = renderer
      .create(
        <BrowserRouter>
          {' '}
          <Nav />
          {' '}
        </BrowserRouter>,
      )
      .toJSON();
    expect(nav).toMatchSnapshot();
  });
});
