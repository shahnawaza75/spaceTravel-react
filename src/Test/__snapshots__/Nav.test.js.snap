// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the navigator Navigation is rendered 1`] = `
Array [
  " ",
  <ul
    className="nav-items"
  >
    <li>
      <a
        aria-current={null}
        href="/rockets"
        onClick={[Function]}
        style={
          Object {
            "color": "blue",
            "textDecoration": "none",
          }
        }
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/missions"
        onClick={[Function]}
        style={
          Object {
            "color": "blue",
            "textDecoration": "none",
          }
        }
      >
        Missions
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/profile"
        onClick={[Function]}
        style={
          Object {
            "color": "blue",
            "textDecoration": "none",
          }
        }
      >
        My Profile
      </a>
    </li>
  </ul>,
  " ",
]
`;
